src/compiler.rs

```diff
@@ -144,9 +159,48 @@ impl AmberCompiler {
         format!("{}\n{}", header, res)
     }
 
+    pub fn document(&self, block: Block, meta: ParserMetadata, output: String) {
+        let base_path = PathBuf::from(meta.get_path().expect("Input file must exist in docs generation"));
+        let base_dir = fs::canonicalize(base_path)
+            .map(|val| val.parent().expect("Parent dir must exist in docs generation").to_owned().clone());
+        if let Err(err) = base_dir {
+            Message::new_err_msg("Couldn't get the absolute path to the provided input file")
+                .comment(err.to_string())
+                .show();
+            std::process::exit(1);
+        }
+        let base_dir = base_dir.unwrap();
+        let ast_forest = self.get_sorted_ast_forest(block, &meta);
+        for (path, block) in ast_forest {
+            let dep_path = {
+                let dep_path = fs::canonicalize(PathBuf::from(path.clone()));
+                if dep_path.is_err() { continue }
+                dep_path.unwrap()
+            };
+            if !dep_path.starts_with(&base_dir) { continue }
```

Comment:
Just a preference, but I like multiline `if`. Exception is when they return something.

And if we're preparing this var in this block, let's prepare it completely :D
```suggestion
            let dep_path = {
                let dep_path = fs::canonicalize(PathBuf::from(path.clone()));
                if dep_path.is_err() {
                    continue
                }
                let dep_path = dep_path.unwrap();

                if !dep_path.starts_with(&base_dir) {
                    continue
                }

                dep_path
            };
```